tests/test_filter_params.py

```diff
@@ -0,0 +1,422 @@
+from dataservice.api.genomic_file.models import GenomicFile
+from dataservice.api.sequencing_experiment.models import SequencingExperiment
+from dataservice.api.sequencing_center.models import SequencingCenter
+from dataservice.api.family_relationship.models import FamilyRelationship
+from dataservice.api.study_file.models import StudyFile
+from dataservice.api.cavatica_app.models import CavaticaApp
+from dataservice.api.cavatica_task.models import (
+    CavaticaTask,
+    CavaticaTaskGenomicFile
+)
+
+# pytest_plugins = ['tests.mocks']
+
+ENTITY_TOTAL = 15
+SAMPLE_DATE = str(datetime.now().replace(tzinfo=tz.tzutc()))
+
+ENTITY_ENDPOINT_MAP = {
+    Study: '/studies',
+    Investigator: '/investigators',
+    StudyFile: '/study-files',
+    Family: '/families',
+    # FamilyRelationship: '/family-relationships',
+    CavaticaApp: '/cavatica-apps',
+    SequencingCenter: '/sequencing-centers',
+    Participant: '/participants',
+    Diagnosis: '/diagnoses',
+    Phenotype: '/phenotypes',
+    Outcome: '/outcomes',
+    Biospecimen: '/biospecimens',
+    GenomicFile: '/genomic-files',
+    SequencingExperiment: '/sequencing-experiments',
+    CavaticaTask: '/cavatica-tasks'
+    # CavaticaTaskGenomicFile: '/cavatica-task-genomic-files'
+}
+
+ENTITY_PARAMS = {
+    'fields': {
+        '/studies': {'external_id': 'Study_0'},
+        '/investigators': {'name': 'Investigator_0'},
+        '/study-files': {'file_name': 'File_0'},
+        '/cavatica-apps': {'name': 'App_0'},
+        '/families': {'external_id': 'Family_0'},
+        '/sequencing-centers': {'name': ''},
+        '/participants': {
+            'is_proband': True,
+            'consent_type': 'GRU-IRB',
+            'race': 'asian',
+            'ethnicity': 'not hispanic',
+            'gender': 'male'
+        },
+        '/biospecimens': {'analyte_type': 'DNA',
+                          'external_sample_id': 'Biospecimen_0'},
+        '/diagnoses': {'diagnosis_category': 'birth defect',
+                       'source_text_diagnosis': 'Diagnosis_0'},
+        '/outcomes': {'external_id': 'Outcome_0',
+                      'vital_status': 'alive'},
+        '/phenotypes': {'external_id': 'Phenotype_0'},
+        '/sequencing-experiments': {
+            'external_id': 'SeqExp_0',
+            'experiment_strategy': 'WXS',
+            'library_name': 'Library_1',
+            'library_strand': 'Unstranded',
+            'is_paired_end': False,
+            'platform': 'Platform_0'
+        },
+        '/genomic-files': {
+            'external_id': 'GenomicFile_0',
+            'file_name': 'hg38.fq',
+            'data_type': 'reads',
+            'file_format': 'fastq',
+            'size': 1000,
+            'urls': ['s3://bucket/key'],
+            'hashes': {'md5': str(uuid.uuid4())},
+            'controlled_access': False
+        },
+        '/cavatica-tasks': {'name': 'CavaticaTask_0'}
+
+    },
+    'filter_params': {
```

Comment:
Can't this be constructed automatically?

Reply:
Not sure what you mean?

Reply:
I mean you could generate a dict of endpoints with their fields based on the above dictionary.

Reply:
Maybe, but I want to tightly control the set of filter_params for each endpoint, so I'm manually defining the dict